x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -672,6 +672,14 @@ export class RulesClient {
   }
 
   public async delete({ id }: { id: string }) {
+    return await retryIfConflicts(
```

Comment:
Note that I will not be backporting this specific change to 7.x since 7.x should still have the SO type as `single`.

Reply:
Do we need this change and the removal code below? I thought the removal of the below code solved the issue?

Reply:
This fixes the "expected 204, got 409" flaky tests. When I unskipped the test suite that was fixed by removing the dead code, I got a bunch of flaky test failures related to the 204/409 issue so I handled both in the same PR

Reply:
Is there any concern with this code change for normal use (outside of tests)? Is it possible/a good idea to move this retry logic into the test suites themselves to avoid exposing any new issue by changing the non test code?

Reply:
This retry logic is already used in the rules client update and updateApiKey functions, which both have the possibility of returning a conflict if multiple Kibanas are updating a rule at the same time. We have to add this now to the delete function for `multiple-isolated` alert SOs because delete is not just deleting the SO doc with id `${spaceId}:alert:${alertId}`. Instead it could be updating an alert SO that is shared between spaces by removing one of the spaces from the `namespaces` field.

Reply:
To be clear, the "expected 204, got 409" errors are our functional tests correctly telling us that we've introduced a race condition with the change from `single` to `multiple-isolated` for the alert SO type. Yay for tests!

Reply:
Thanks for the explanation, makes sense to me!